garage/tf/models/gaussian_mlp_model.py

```diff
@@ -0,0 +1,190 @@
+            mean_model = MLPModel(
+                input_var=input_var,
+                output_dim=self._output_dim,
+                hidden_sizes=self._hidden_sizes,
+                hidden_nonlinearity=self._hidden_nonlinearity,
+                output_nonlinearity=self._output_nonlinearity,
+                scope=self._scope + "/mean_network")
+            mean_var = mean_model.outputs[0]
+
+            if self._adaptive_std:
+                b = tf.constant_initializer(self._init_std_param)
+                std_model = MLPModel(
+                    input_var=input_var,
+                    output_dim=self._output_dim,
+                    hidden_sizes=self._std_hidden_sizes,
+                    hidden_nonlinearity=self._std_hidden_nonlinearity,
+                    output_nonlinearity=self._std_output_nonlinearity,
+                    output_b_init=b,
+                    scope=self._scope + "/std_network")
+
+                std_param_var = std_model.outputs[0]
+            else:
+                p = tf.constant_initializer(self._init_std_param)
+                std_param_var = ParameterLayer(
+                    length=self._output_dim,
+                    initializer=p,
+                    trainable=self._learn_std,
+                    name=self._scope + "/std_network")(input_var)
+
+        if self._min_std_param:
+            std_param_var = Lambda(
+                lambda x, f, p: f(x, p),
+                arguments={
+                    "f": tf.maximum,
+                    "p": self._min_std_param
+                },
+                name=self._scope + "/std_min")(std_param_var)
+        if self._max_std_param:
+            std_param_var = Lambda(
+                lambda x, f, p: f(x, p),
+                arguments={
+                    "f": tf.minimum,
+                    "p": self._max_std_param
+                },
+                name=self._scope + "/std_max")(std_param_var)
+
+        if self._std_parameterization == "exp":
+            std_var = Lambda(
+                lambda x, f: f(x),
+                arguments={'f': tf.exp},
+                name=self._scope + "/std_parameterization")(std_param_var)
+        elif self._std_parameterization == "softplus":
+            std_var = Lambda(
+                lambda x, f, g: f(1 + g(x)),
+                arguments={
+                    "f": tf.log,
+                    "g": tf.exp
+                },
+                name=self._scope + "/std_parameterization")(std_param_var)
+        else:
+            raise NotImplementedError
+
+        sample_var = Lambda(
+            lambda x, f, s: f(x[0], x[1]).sample(seed=s),
+            arguments={
+                "f": tf.contrib.distributions.MultivariateNormalDiag,
+                "s": ext.get_seed()
+            },
+            name=self._scope + "/sample_var")([mean_var, std_param_var])
+
+        return Model(
+            inputs=input_var,
+            outputs=[mean_var, std_var, std_param_var, sample_var])
```

Comment:
I think you will also need outputs for important distribution parameters such as entropy() or logpdf() (or some other way of accessing those, such as providing an output for the entire distribution).